ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java

```diff
@@ -115,24 +132,36 @@ private void registerForCreateAs(final ConfiguredStatement<? extends CreateAsSel
         ));
 
     registerSchema(
-        queryMetadata.getLogicalSchema(),
+        queryMetadata.getLogicalSchema().key(),
+        queryMetadata.getResultTopic().getKafkaTopicName(),
+        queryMetadata.getResultTopic().getKeyFormat().getFormatInfo(),
+        queryMetadata.getPhysicalSchema().keySchema().features(),
+        cas.getSessionConfig().getConfig(false),
+        cas.getStatementText(),
+        true,
+        KsqlConstants.SCHEMA_REGISTRY_KEY_SUFFIX
+    );
+    registerSchema(
+        queryMetadata.getLogicalSchema().withoutPseudoAndKeyColsInValue().value(),
```

Comment:
minor: consider adding a `registerSchemas` method, which takes the logical schema and both formats. Internally, it justs called `registerSchema` for key then value.

This would make the code easier to read and remove a bit of duplication.